### PR TITLE
feat: Added getStructuredDataScript util

### PR DESCRIPTION
**Pull request by ambroisemaupate**

### CODE REVIEW

File: utils/structured-data.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export function getStructuredDataScript(content: MaybeRef<structuredDataContent>) {
+    const jsonLdContent = {
+        '@context': 'https://schema.org',
+        ...removeNullishNestedKeys(toValue(content)),
+    }
+
+    return h('script', {
+        type: 'application/ld+json',
+        innerHTML: toDisplayString(jsonLdContent),
+    })
```
Comment (manuelodelain) [inline utils/structured-data.ts:32]: Why not using `useHead` for injecting script into the body for example? This util could be a composable like `useStructuredData()`

File: utils/structured-data.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { h, toDisplayString } from 'vue'
+import type { MaybeRef } from 'vue'
+
+type structuredDataContent = Record<string, unknown>
+
+const isNullishValue = (v: unknown) => {
+    if (typeof v === 'number') return false
+    return v === null || v === undefined || !Object.keys(v).length || (Array.isArray(v) && !v.length)
+}
```
Comment (manuelodelain) [inline utils/structured-data.ts:9]: I think it should be split into another util file, e.g. `/util/null.ts` (group with other utils for nullish values) or `/util/is-nullish-value.ts`

File: utils/structured-data.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    Object.keys(obj).forEach((key) => {
+        const value = obj[key]
+
+        if (isNullishValue(value)) delete obj[key]
+        else if (Array.isArray(value) && value.every((v) => isNullishValue(v))) delete obj[key]
+        else if (value && typeof value === 'object') removeNullishNestedKeys(value as structuredDataContent)
+    })
+
+    return obj
+}
```
Comment (manuelodelain) [inline utils/structured-data.ts:21]: Same as `isNullishValue()`, i.e. split into another file for reuse

File: server/plugins/cache-control.ts
Comment (manuelodelain) [inline server/plugins/cache-control.ts:1]: The plugin will be automatically loaded and executed by Nuxt. Even if we don't need it. I think plugin into layer is a bit tricky. Maybe we should use a module for this kind of thing? Like a Nuxt module `@rezo-zero/nuxt-cache-control` who injects the plugin? Another solution would be to nest the plugin into a folder like `plugins/cache/cache-controls` (not auto loaded) and we could add it manually into the Nuxt config.

Comment (ambroisemaupate) [reply server/plugins/cache-control.ts:1]: I think a nuxt-module would be preferable. 

Every project using a Cache Reverse-Proxy (Varnish, Cloudflare, Vercel) needs to define `cache-control` header on SSR responses.

**Pull request merged**